kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/InternalRequestFrame.java

```diff
@@ -38,4 +40,9 @@ public Filter recipient() {
     public CompletableFuture<?> promise() {
         return promise;
     }
+
+    @Override
+    public DecodedResponseFrame<?> responseFrame(ResponseHeaderData header, ApiMessage message) {
+        return new InternalResponseFrame<>(recipient, apiVersion, correlationId, header, message, promise);
```

Comment:
ditto - consider failing if apiKey() != message.apiKey(). Maybe add a unit test for InternalRequestFrame

Reply:
done, reworked it so that that assertion is solely contained in `DecodedRequestFrame` and then it calls a protected method to create the frame.